### ISSUE TITLE
MSP430 example adaptation does not work. 

### ISSUE THREAD

**Issue by deadeert** [bdcht/amoco#76]
Hello,

The example provided at http://amoco.readthedocs.io/en/latest/quickstart.html#loading-binary-data works for x86 bytecode but generates an error for a MSP430 bytecode. 

```
>>>shellcode = (b"\xeb\x16\x5e\x31\xd2\x52\x56\x89\xe1\x89\xf3\x31\xc0\xb0\x0b\xcd")
>>>p  = amoco.system.loader.load_program(shellcode)
>>>from amoco.arch.x86 import cpu_x86
>>>p.cpu = cpu_x86
>>>z=amoco.lsweep(p)
>>>z.getblock(0)
<block object (0x0) at 0x7f8038af90a0>
 >>>b=_
 >>>print(b.view)
# --- block 0x0 -------------------------------
0x0                 'eb16'    jmp         *0x18

```

When using the same method for MSP430 arch, an error is raised : 
```

>>> by=b"\x15\x42\x5c\x01" #mov	&0x015c, r5
>>> p  = amoco.system.loader.load_program(by)
amoco.system.loader: WARNING: unknown format
amoco.system.raw: WARNING: a cpu module must be imported
>>> from amoco.arch.msp430 import cpu
>>> p.cpu = cpu
>>> z=amoco.lsweep(p)
>>> z.getblock(0)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    z.getblock(0)
  File "/usr/lib/python3.6/site-packages/amoco-2.5.1-py3.6.egg/amoco/main.py", line 120, in getblock
    b = next(ib)
  File "/usr/lib/python3.6/site-packages/amoco-2.5.1-py3.6.egg/amoco/main.py", line 88, in iterblocks
    for i in seq:
  File "/usr/lib/python3.6/site-packages/amoco-2.5.1-py3.6.egg/amoco/main.py", line 66, in sequence
    i = p.read_instruction(loc)
  File "/usr/lib/python3.6/site-packages/amoco-2.5.1-py3.6.egg/amoco/system/core.py", line 534, in read_instruction
    i = self.cpu.disassemble(istr[0],**kargs)
  File "/usr/lib/python3.6/site-packages/amoco-2.5.1-py3.6.egg/amoco/arch/core.py", line 214, in __call__
    i = s.decode(bytestring,e,i=self.__i,ival=b.ival,iclass=self.iclass)
  File "/usr/lib/python3.6/site-packages/amoco-2.5.1-py3.6.egg/amoco/arch/core.py", line 489, in decode
    self.hook(obj=i,**kargs)
  File "/usr/lib/python3.6/site-packages/amoco-2.5.1-py3.6.egg/amoco/arch/msp430/spec_msp430.py", line 71, in msp430_doubleop
    src,data = getopd(obj,As,Sreg,data,CGR=True)
  File "/usr/lib/python3.6/site-packages/amoco-2.5.1-py3.6.egg/amoco/arch/msp430/spec_msp430.py", line 24, in getopd
    return r[0:size],data
TypeError: 'int' object is not subscriptable

```

Thanks for the support!  

Cheers

**Comment by deadeert** [comment]
More over, I found another bug based on your discussion with Mr PwnTools (https://github.com/bdcht/amoco/issues/10). 

 ```
 1 import amoco
  2 from amoco.arch.msp430 import cpu as cpu
  3
  4
  5 bc=b"\x31\x40\x00\x44"
  6 loc=cpu.cst(0,16)
  7 instr=[]
  8 while len(bc)>0:
  9   i = cpu.disassemble(bc,address=loc)
 10   l = i.length
 11   instr.append(i)
 12   bc = bc[l:]
 13   loc+=l
 14   b=amoco.code.block(instr)
 15
 16
 17 print(b.view)

Traceback (most recent call last):
  File "/tmp/test-amoco.py", line 17, in <module>
    print(b.view)
  File "/usr/lib/python3.6/site-packages/amoco-2.5.1-py3.6.egg/amoco/ui/views.py", line 78, in __str__
  File "/usr/lib/python3.6/site-packages/amoco-2.5.1-py3.6.egg/amoco/ui/views.py", line 61, in _vltable
  File "/usr/lib/python3.6/site-packages/amoco-2.5.1-py3.6.egg/amoco/arch/core.py", line 131, in toks
TypeError: MSP430_synthetic() got an unexpected keyword argument 'toks'

```
Note that 
```
print(b)
```  
Triggers the same issue. 

I did not create a  new issue because I think it is linked. 

Let me know if I can help (running some test case for you). 

Cheers,

**Comment by bdcht** [comment]
Hi, thanks for the reports. I'll dig into these 2 issues tonight, msp430 lacks some basic testing anyway...stay tuned.

**Comment by deadeert** [comment]
Hello,

No rush and thanks for the support ! 
Let me know if I can help.

**Comment by bdcht** [comment]
fixed by commit a06a520

**Comment by deadeert** [comment]
Thanks for the commit will try this afternoon. Cheers
